pyfr/plugins/tavg.py

```diff
@@ -358,3 +363,200 @@ def __call__(self, intg):
+                              in self.idxs]
+
+                self.mapping = [(amap, smap)]
+                self.anames = sorted(anames)
+                self.snames = sorted(snames)
+
+                # Initialise std max and avg registers
+                nsnames = len(snames)
+                if self.std_all:
+                    self.std_max = np.zeros(nsnames)
+                    self.std_avg = np.zeros(nsnames)
+
+                # Build fun avg expressions
+                c = cfg.items_as('constants', float)
+                self.fexprs = fexprs = []
+                for k in cfg.items(cfgsect, prefix='fun-avg-'):
+                    fexprs.append(cfg.getexpr(cfgsect, k, subs=c))
+            else:
+                self.mapping.append((amap, smap))
+
+            # Check for compatibility of files
+            if self.uuid != file['mesh_uuid']:
+                raise RuntimeError('Average files computed on different meshes')
+            if self.region != cfg.get(cfgsect, 'region'):
+                raise RuntimeError('Average files computed on different regions')
+
+            for k in cfg.items(cfgsect, prefix='avg-'):
+                if cfg.get(cfgsect, k) != self.cfg.get(self.cfgsect, k):
+                    raise RuntimeError('Different average field definitions')
+
+    def _merge_data(self, outf):
+        avg_time_inv = 1/self.avg_time
+
+        (file0, stats0, cfg0, dt0), f = self._copy_pop(self.files)
+        (amap0, smap0), mapping = self._copy_pop(self.mapping)
+
+        for key in self.dkeys:
+            # Initialise accumulators
+            data = file0[key]
+            avg_acc = data[:, amap0]*dt0
+            if self.std_all:
+                var_acc = data[:, smap0]**2*dt0
+
+            # Perform the rest of the accumulations
+            for (file, stats, cfg, dt), (amap, smap) in zip(f, mapping):
+                data = file[key]
+
+                avg_acc += data[:, amap]*dt
+                if self.std_all:
+                    var_acc += data[:, smap]**2*dt
+
+            if self.fexprs:
+                avg = avg_acc.transpose(1, 0, 2)*avg_time_inv
+                fun_avg = self._eval_fun_exprs(avg)
+                outf[key] = np.vstack((avg, fun_avg)).transpose(1, 0, 2)
+            else:
+                outf[key] = avg_acc*avg_time_inv
+
+            if self.std_all:
+                std = np.sqrt(var_acc*avg_time_inv).transpose(1, 0, 2)
+                np.maximum(self.std_max, np.amax(std, axis=(1, 2)),
+                           out=self.std_max)
+                self.std_avg += np.mean(std, axis=(1, 2))
+
+    def _merge_meta(self, outf):
+        cfgsect = self.stats.get('tavg', 'cfg-section')
+        self.stats.set('tavg', 'merged-from', self.merged_from)
+        self.stats.set('tavg', 'range', self.merged_range)
+
+        # If all files have full std stats then these can be writen
+        if self.std_all:
```

Comment:
Check these stats by running a toy simulation with two plugin instances: one continuous writing every 2*dt and another windowed writing every dt and verify that merged windowed ~ continuous.

Reply:
Just following up here.